src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Helpers.cs

```diff
@@ -137,8 +138,46 @@ public void WriteLine(string source)
                 _sb.AppendLine(source);
             }
 
+            public void WriteBlock(string source)
+            {
+                foreach (string value in source.Split('\n'))
+                {
+                    string line = value.Trim();
+                    switch (line)
+                    {
+                        case "{":
```

Comment:
Interesting, presumably you do this to track indentation when partial blocks are being emitted?
```C#
writer.WriteBlock("""
      while (true)
      {
      """);
```
I generally assumed users would only use complete blocks in multi-line literals so this type of tracking is not necessary.

Reply:
That's right, just that they'd also need to figure out/keep track of the correct indentation for their block, and possibly recalculate it when code is refactored.

This emitter implementation has quite a few reused sub-emitters (i.e. the various `Emit*Logic` methods) all with different indentation levels. Per https://github.com/dotnet/runtime/pull/83614#discussion_r1140664498, the way the writer is set up now allows users to not have to keep track of the indentation themselves. For me, this is a non-trivial convenience, though I'm not sure how other folks would feel about it.

Reply:
> presumably you do this to track indentation when partial blocks are being emitted?

No I do it so I don't have to keep track of the correct indentation indentation for each block. This emitter implementation has quite a few sub-emitters (i.e. the various `Emit*Logic` methods) with different indentation levels. They are all enveloped in multiple containing call sites/methods. Having the writer handle the indentation makes things a bit easier.